pkg/build/builder/sti.go

```diff
@@ -253,6 +253,9 @@ func (s *S2IBuilder) Build() error {
 
 	cname := containerName("s2i", s.build.Name, s.build.Namespace, "post-commit")
 	if err := execPostCommitHook(s.dockerClient, s.build.Spec.PostCommit, buildTag, cname); err != nil {
+		// update the buildobject with the failure reason
+		s.build.Status.Reason = api.StatusReasonPostCommitHookFailed
+		updateBuildStatus(s.client, s.build)
```

Comment:
if you rename this to updateBuildFailureReason, then you can also change it to take args like:
`updateBuildStatusFailureReason(s.client,s.build,api.StatusReasonPostCommitHookFailed)`

which seems cleaner than what you're currently doing.